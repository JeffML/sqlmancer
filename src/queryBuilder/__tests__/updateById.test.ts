import { withDialects } from './__utilties__'
import { ActorUpdateByIdBuilder } from './__fixtures__/models'

describe('UpdateManyBuilder', () => {
  withDialects((options, rollback) => {
    describe('basic queries', () => {
      test('no additional options', async () => {
        const builder = new ActorUpdateByIdBuilder(options, 10, { firstName: 'YENNEFER' })
        const { sql, bindings } = builder.toQueryBuilder().toSQL()
        await rollback(builder, result => expect(result).toBeBoolean())
        expect(sql).toMatchSnapshot()
        expect(bindings).toMatchSnapshot()
      })

      test('non-existent field', async () => {
        const builder = new ActorUpdateByIdBuilder(options, 10, {
          firstName: 'YENNEFER',
          lastNam: 'OF VENGERBERG',
        } as any)
        const { sql, bindings } = builder.toQueryBuilder().toSQL()
        await rollback(builder, result => expect(result).toBeBoolean())
        expect(sql).toMatchSnapshot()
        expect(bindings).toMatchSnapshot()
      })
    })
  })
})
