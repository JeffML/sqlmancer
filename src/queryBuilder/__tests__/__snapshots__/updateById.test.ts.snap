// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateManyBuilder pg basic queries no additional options 1`] = `"update \\"actor\\" set \\"first_name\\" = ? where \\"actor\\".\\"actor_id\\" = ?"`;

exports[`UpdateManyBuilder pg basic queries no additional options 2`] = `
Array [
  "YENNEFER",
  10,
]
`;

exports[`UpdateManyBuilder pg basic queries non-existent field 1`] = `"update \\"actor\\" set \\"first_name\\" = ? where \\"actor\\".\\"actor_id\\" = ?"`;

exports[`UpdateManyBuilder pg basic queries non-existent field 2`] = `
Array [
  "YENNEFER",
  10,
]
`;
